osquery/tables/system/windows/drivers.cpp

```diff
@@ -159,42 +168,63 @@ std::string getDriverImagePath(const std::string& service_key) {
     return "";
   }
 
-  // Unify the image path as systemRoot can contain systemroot and/or system32
-  std::transform(path.begin(), path.end(), path.begin(), ::tolower);
+  return kNormalizeImage(path);
+}
 
-  char systemRoot[MAX_PATH] = {0};
-  GetSystemDirectory(systemRoot, MAX_PATH);
+Status genServiceKeyMap(
+    std::map<std::string, std::string>& services_image_map) {
+  // Attempt to get all of the services image paths in the default location
+  SQL sql("SELECT key, data FROM registry WHERE path LIKE'" + kServiceKeyPath +
+          "%\\ImagePath'");
 
-  return systemRoot +
-         boost::regex_replace(path, boost::regex("^.*?system32"), "");
+  // Something went wrong
+  if (sql.rows().empty()) {
+    return Status::failure("Failed to retrieve services image path cache");
+  }
+
+  char sys_root[MAX_PATH] = {0};
+  GetSystemDirectory(sys_root, MAX_PATH);
+  for (auto& row : sql.rows()) {
+    if (row.count("key") == 0 || row.count("data") == 0) {
+      continue;
+    }
+    services_image_map[row.at("key")] = kNormalizeImage(row.at("data"));
```

Comment:
There are two lookups for each key. You could do it with one using the `find` and iterators. Would you like to?
Smth like:
```c++
auto keyIt = row.find("key");
auto dataIt = row.find("data");
if (keyIt != row.end() && dataIt != row.end()) {
  services_image_map[keyIt->second] = kNormalizeImage(dataIt->second);
}
```